### PR TITLE
GH-44878: [Archery] Only suppress Docker progress logs when running on CI

### PR DESCRIPTION
**Pull request by pitrou**
### Rationale for this change

https://github.com/apache/arrow/pull/44669 went overboard by always suppressing Docker progress output. This may be desirable on CI, to avoid flooding the logs, but is annoying when running `archery docker` locally.

### What changes are included in this PR?

Only suppress Docker progress output if on CI.

### Are these changes tested?

Yes.

### Are there any user-facing changes?

Yes, this restores the previous behavior for non-CI runs.

* GitHub Issue: #44878

### CODE REVIEW

**Comment by pitrou** [comment]
Also cc @kszucs

File: dev/archery/archery/docker/tests/test_docker.py [hunk 1]
```diff
@@ -217,6 +217,15 @@ def arrow_compose_path(tmpdir):
     return create_config(tmpdir, arrow_compose_yml, arrow_compose_env)
 
 
+# Make sure that the tests behave the same on CI as when run locally.
+@pytest.fixture(autouse=True)
+def no_ci_env_variables(monkeypatch):
+    """Remove requests.sessions.Session.request for all tests."""
```
Comment (kou) [inline dev/archery/archery/docker/tests/test_docker.py:223]: Is this docstring correct?

**Review by kou — APPROVED** [review]
+1

File: dev/archery/archery/docker/tests/test_docker.py (hunk 1 above)
Comment (pitrou) [reply dev/archery/archery/docker/tests/test_docker.py:223]: Definitely not, sorry. Will fix.

**Comment by pitrou** [comment]
@github-actions crossbow submit test-conda-cpp

**Pull request merged**